release.go

```diff
@@ -248,11 +249,34 @@ func buildReleaseCommits(ctx context.Context, ghClient *githubClient, commits []
 			return nil, err
 		}
 		for i := range v {
+			sha := *v[i].MergeCommitSHA
+			shaPRs[sha] = v[i]
 			number := *v[i].Number
-			prs[number] = v[i]
+			numPRs[number] = v[i]
 		}
 	}
 
+	getPullRequest := func(commit Commit) (*github.PullRequest, error) {
+		if !commit.IsMerge() {
+			return nil, nil
+		}
+		if pr, ok := shaPRs[commit.Hash]; ok {
+			return pr, nil
+		}
+		prNumber, ok := commit.PullRequestNumber()
+		if !ok {
+			return nil, nil
+		}
+		if pr, ok := numPRs[prNumber]; ok {
+			return pr, nil
+		}
+		pr, err := ghClient.getPullRequest(ctx, event.Owner, event.Repo, prNumber)
```

Comment:
If the PR is not retrieved on the first fetch, the API will be called as usual using the PR Number.